src/ulistener.rs

```diff
@@ -0,0 +1,223 @@
+/********************************************************************************
+ * Copyright (c) 2024 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ********************************************************************************/
+
+use crate::{UMessage, UStatus};
+
+pub trait UListener {
+    fn on_receive(&self, message: Result<UMessage, UStatus>);
+}
+
+pub trait NotificationUListener: UListener + Send + Sync {}
+
+pub trait PublishUListener: UListener + Send + Sync {}
+
+pub trait RequestUListener: UListener + Send + Sync {}
+
+pub trait ResponseUListener: UListener + Send + Sync {}
+
+pub trait GenericUListener: UListener + Send + Sync {
+    fn is_match(&self, message: UMessage) -> bool;
```

Comment:
Is there any value in having this function? If the transport needs to pass in the message, it can as well simply invoke `onMessage(UMessage)` and let the listener decide if it wants to process or not, or am I mistaken?

Reply:
Ah... that's true I think!